src/__tests__/App.test.tsx

```diff
@@ -1,19 +1,41 @@
-import { shallow } from 'enzyme'
+import { render, screen } from '@testing-library/react'
 import React from 'react'
 import { Provider } from 'react-redux'
-import configureStore from 'redux-mock-store'
+import createMockStore from 'redux-mock-store'
+import thunk from 'redux-thunk'
 
 import App from '../App'
+import { RootState } from '../shared/store'
 
-it('renders without crashing', () => {
-  const mockStore = configureStore()({})
+const mockStore = createMockStore<RootState, any>([thunk])
 
-  const AppWithStore = () => (
-    <Provider store={mockStore}>
+it('renders without crashing', async () => {
+  // Supress the console.log in the test ouput
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  jest.spyOn(console, 'log').mockImplementation(() => {})
+
+  const store = mockStore({
+    components: {
+      sidebarCollapsed: false,
+    },
+    breadcrumbs: {
+      breadcrumbs: [],
+    },
+    user: {
+      permissions: [],
+    },
+  } as any)
+
+  render(
+    <Provider store={store}>
       <App />
-    </Provider>
+    </Provider>,
   )
 
-  const wrapper = shallow(<AppWithStore />)
-  expect(wrapper).toBeDefined()
+  expect(
+    await screen.findByRole('heading', { name: /dashboard\.label/i }, { timeout: 8000 }),
+  ).toBeInTheDocument()
+
+  // eslint-disable-next-line no-console
+  ;(console.log as jest.Mock).mockRestore()
```

Comment:
Earlier in the test, `console.log` is mocked to keep the console clean. Then, since `console.log` is now a jest mock function, we need to cast it as `jest.Mock` so that we can restore it.

Happy to have suggestions on a better way :smile:

Reply:
All good thanks, just wasn't sure I understood line 40, I guess the syntax threw me off a bit, 😇

Reply:
Yeah, the leading semicolon just came out that way due to eslint... Otherwise it thought the cast was a function call chained to the return value of `.toBeInTheDocument` - which isn't the intent at all.